styx-scheduler-service/src/main/java/com/spotify/styx/ServiceAccountKeyManager.java

```diff
@@ -43,6 +44,33 @@ public ServiceAccountKey createP12Key(String serviceAccount) throws IOException
         .setPrivateKeyType("TYPE_PKCS12_FILE"));
   }
 
+  public boolean serviceAccountExists(String serviceAccount) throws IOException {
+    try {
+      iam.projects().serviceAccounts().get("projects/-/serviceAccounts/" + serviceAccount)
+          .execute();
+      return true;
+    } catch (GoogleJsonResponseException e) {
+      if (e.getStatusCode() == 404) {
+        return false;
+      }
+      throw e;
```

Comment:
There was a discussion that `Throwables.propagate` has been deprecated by Google. Also in this case we don't need a `RuntimeException` I think.